src/workerd/jsg/url.c++

```diff
@@ -394,4 +401,1853 @@ kj::Maybe<UrlSearchParams::EntryIterator::Entry> UrlSearchParams::EntryIterator:
   };
 }
 
+// ======================================================================================
```

Comment:
was this code moved from somewhere? reviewing it will be hard...

Also, maybe `url-pattern.c++`?

Reply:
Yes and no. This is a majorly refactored implementation of the original api/urlpattern.c++ implementation, updated to use kj::String (instead of jsg::UsvString) and jsg::Url instead of the original api/url-standard.c++ impl. It's changed significantly enough that it is safer to treat it as new code as opposed to moved code.

Reply:
The reason I kept this in `jsg/url.++` and not a separate `jsg/url-pattern.c++` is really just to limit the number of places where we touch ada-url includes directly and keeping all of the URL-related functionality together. If you feel strongly about it, I could separate this out but I'd just generally prefer not to.